### ISSUE TITLE
Rule Option Proposal: `no-eval` allows indirect-evals

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#4441]
Spawned from #4399 

Indirect `eval` calls are not danger relatively than direct `eval` calls.
So I propose an option to allow indirect `eval` calls.

```js
{
    "no-eval": [2, {"allowIndirect": true}] // default is false
}
```

This option considers the following patterns are problem:

```js
eval("var a = 0");
```

This option considers the following patterns are NOT problem:

```js
(0, eval)("var a = 0");

function foo(eval) {
    eval("var a = 0");
}
foo(eval);
```

I'm happy to work on this.

**Comment by michaelficarra** [comment]
For the second example (`function foo ...`), it should be stated that this is a direct eval, but we just can't catch it. So this is also not reported if `allowIndirect` is `false`.

**Comment by mysticatea** [comment]
Wow... `function foo(eval) { eval("var a = 0"); }` is a direct eval?

**Comment by michaelficarra** [comment]
Yep. See https://github.com/eslint/eslint/issues/4399#issuecomment-156181448.

**Comment by mysticatea** [comment]
Ah... sorry, thank you.
Hmm.
